File: Perl/scripts/Problem004.pl
use List::Util qw(max);

sub isPallindrome {
    my $n = @_[0]; my $k = reverse $n;
    return  $k == $n;
}

$start = time(); @palls = ();

for($i = 999; $i >= 100; $i--) {
    for($j = 999; $j >= 100; $j--) {
        if (isPallindrome($i*$j)) {
            push(@palls, $i*$j);
        }
    }
}

$ans = max(@palls);
$exc_time = time() - $start;
print "\nAnswer: $ans\n";
print "Time Taken: $exc_time seconds\n\n";

File: Python/scripts/Problem004.py
from time import time

isPall = lambda s : str(s)==str(s)[::-1]
palls = []
start = time()

for i in range(999, 100, -1):
    for j in range(999, 100, -1):
        if isPall(i*j):
            palls.append(i*j)

ans = max(palls)

if __name__ == "__main__":
    print(f"\nAnswer: { ans }")
    print(f"Time taken: { time() - start }\n")
